### ISSUE TITLE
Script checks SAP profiles which are not relevant.

### ISSUE THREAD

**Issue by lpalovsky** [scmschmidt/wmp_check#1]
Hello. 
During the WMP basic test we encountered false positive warnings from wmp_check.
Looks like the the pattern search includes files in sap profile path which are not relevant for wmp setup. 

First example: 
`[NOTE]  No entry for the WMP capture program found in/usr/sap/FED/SYS/profile/FED_ERS13_x1c215.lst.`
 As you can see the check included file FED_ERS13_x1c215.lst. Not sure where it came from but profile with *.lst suffix should not be checked.

Second example:
```
[NOTE]  WMP entry for the WMP capture program found for instance FED_D10_x1c215.1.
[NOTE]  WMP entry for the WMP capture program found for instance FED_D10_x1c215.2.
[NOTE]  WMP entry for the WMP capture program found for instance FED_D10_x1c215.3.
```

Pattern search includes as well backed up profiles created by NW instance after changing parameters via RZ10 transaction.
Those profiles are not active and might contain old configuration without the wmp capture line, which would trigger false positve error message.  

Third example:
Don't have an output log, but pattern included files that have valid profile name with some prefix like `backup_FED_D10_x1c215`

Maybe the search pattern should be changed to something like: `^<SID>_<INST>_<hostname>$`

**Comment by scmschmidt** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

Hi Lumir,

Thanks a lot! 

I suspected that tuning the pattern would be necessary. I'll look into 
this week (maybe on Wednesday).

Sören


Sören Schmidt 
SAP Solution Architect

SUSE Software Solutions Germany GmbH
Maxfeldstr. 5
90409 Nürnberg
Germany
 
(HRB 36809, AG Nürnberg)
Geschäftsführer: Felix Imendörffer
 

Am Montag, den 07.12.2020, 04:41 -0800 schrieb Lumir Palovsky:
> Hello.
> During the WMP basic test we encountered false positive warnings from
> wmp_check.
> Looks like the the pattern search includes files in sap profile path
> which are not relevant for wmp setup.
> First example:
> [NOTE] No entry for the WMP capture program found
> in/usr/sap/FED/SYS/profile/FED_ERS13_x1c215.lst.
> As you can see the check included file FED_ERS13_x1c215.lst. Not sure
> where it came from but profile with *.lst suffix should not be
> checked.
> Second example:
> [NOTE]  WMP entry for the WMP capture program found for instance
> FED_D10_x1c215.1.
> [NOTE]  WMP entry for the WMP capture program found for instance
> FED_D10_x1c215.2.
> [NOTE]  WMP entry for the WMP capture program found for instance
> FED_D10_x1c215.3.
> Pattern search includes as well backed up profiles created by NW
> instance after changing parameters via RZ10 transaction.
> Those profiles are not active and might contain old configuration
> without the wmp capture line, which would trigger false positve error
> message.
> Third example:
> Don't have an output log, but pattern included files that have valid
> profile name with some prefix like backup_FED_D10_x1c215
> Maybe the search pattern should be changed to something like:
> ^<SID>_<INST>_<hostname>$
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.
-----BEGIN PGP SIGNATURE-----

iQIzBAEBCgAdFiEE2XFGYY8a43X0Hi6Ap3Qpht4DsDcFAl/OKBYACgkQp3Qpht4D
sDfCmBAAlAekAuLS0FQoA1rNh61I+ua1T1qPR1dL3d2gqn9fdrizH1QelVmgU0Xb
JF1iGJfscUnAs1RX4UgHVlA0WRNZqwlhtj9Q5QgNNeW5r4d9loKH8OlJo2vDnkSX
6bImxqpnuX8wO+8bVjNoZxyzzHxtZKy+Z3z2++5SxcDUDw2hSHknIEVViJNGfck9
ZUZeuJzNHalmu8I6e0U4ReQYg/YdrGcJMxgoQv8gNkltWwEJ3KjnMvfCn11AkCzO
MGrDFeGKslHQ5fpju3b60cgqCwI5k7shX85KU4zj1eRBMhMLVwL2ytD0OvGa1uEv
QQpMQea7qBPc3nbF1XuMoxaGttvExAFyO2JhzpjCNU3xOHV56InG1ORVvbZ5Yuee
lZqcFct8/Dsflh3qiRZTCKmQx3y/T12Y0KRzIVOQI9psAZxHpnW2TkZPj7L2TpTU
saR6sHm+SiFJo+oMxctWezCoUxXB/iVP8seE0HuX78/PSf1NGsf2kXFv5Ta6DEYf
9yoi2Y7KGxFtDFyINRWySRo1gFW1FzRi7s3B3BF0G151+QgC6JSrge4PWk7JPaz6
dTbA8j/yiQPRPx0emnq65sima5WH70iRG1Un+Lnswa1DJyF+LVYM8DULDZK4200W
NMVpcVmX1n0vw9uLBs2KsIQ3nfdwdik2KIg/KQpRJsDKFPok4NQ=
=xvf9
-----END PGP SIGNATURE-----

**Comment by scmschmidt** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

Hi,

v1.0.3 is available.

I have changed the pattern to: /[[:alnum:]]+_[[:alnum:]]+_[[:alnum:]]+$

For being more narrower, like you suggested, I would need to ask
for/determine the SID. I hope the broader pattern is sufficient for now
.

Thanks for reporting,
Sören

Sören Schmidt 
SAP Solution Architect

SUSE Software Solutions Germany GmbH
Maxfeldstr. 5
90409 Nürnberg
Germany
 
(HRB 36809, AG Nürnberg)
Geschäftsführer: Felix Imendörffer
 

Am Montag, den 07.12.2020, 04:41 -0800 schrieb Lumir Palovsky:
> Hello.
> During the WMP basic test we encountered false positive warnings from
> wmp_check.
> Looks like the the pattern search includes files in sap profile path
> which are not relevant for wmp setup.
> First example:
> [NOTE] No entry for the WMP capture program found
> in/usr/sap/FED/SYS/profile/FED_ERS13_x1c215.lst.
> As you can see the check included file FED_ERS13_x1c215.lst. Not sure
> where it came from but profile with *.lst suffix should not be
> checked.
> Second example:
> [NOTE]  WMP entry for the WMP capture program found for instance
> FED_D10_x1c215.1.
> [NOTE]  WMP entry for the WMP capture program found for instance
> FED_D10_x1c215.2.
> [NOTE]  WMP entry for the WMP capture program found for instance
> FED_D10_x1c215.3.
> Pattern search includes as well backed up profiles created by NW
> instance after changing parameters via RZ10 transaction.
> Those profiles are not active and might contain old configuration
> without the wmp capture line, which would trigger false positve error
> message.
> Third example:
> Don't have an output log, but pattern included files that have valid
> profile name with some prefix like backup_FED_D10_x1c215
> Maybe the search pattern should be changed to something like:
> ^<SID>_<INST>_<hostname>$
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub, or unsubscribe.
-----BEGIN PGP SIGNATURE-----

iQIzBAEBCgAdFiEE2XFGYY8a43X0Hi6Ap3Qpht4DsDcFAl/Q2xcACgkQp3Qpht4D
sDfVNA/9GZgZ2+d8YtNttzTtkF+lPRjewb1CZfQIIrGmpGu6TDmjO2mBCx0yGC1e
m+UPqjcgODDx406aedkvE+Y+knpIa8V9pj33Kbc/XrNYxK4ZbM3p+rX9Iih84NpJ
7MPizTO2J9JV9HFZs1bGtqCgdEbqI4NrgzWCa1MP1q1bSn3y2NMxwSi2l6dhjRtI
32Zvj2yotk7lgq7Vn5lTwylJShm2Qz0XmcJgFISt6MAzEPZN0W6ShW1yrhbL/WOF
uIrei4+WmSLpDkqAKV3Z7GqEFCvfrQjutoYNizJ8pBfIC2+mR3Ic7JX6nnrX4br0
lWGMQd8ex+/pSE6VJyJf597EXWmv5THbE5Q2mivd/WByMBAeilA8+f3uzDiDiPMB
pG1hquCY43+HxBH1WQESNBBVeVWgTHQkxWRNO1fwlGaYdM5YQoFKrMhKKrbZP9u7
b52CjcDRErI8ZGcF1AGCUHuzOG5XZbhpffpTjNucqiFTFPg5v6+Y0fXxFiQSs19W
gDlrAs7GYuiEmjEwwrnUZx7hjIOmQ5TYUT5wUM8eE++oxbfbpjnsVLOO8fgK2B60
813E3OpPva1NOvEGmZn4muPSiU1JM3wGXggCJuJX/0f3kBVPFB3/V2emBbRimrnj
Bcy9q6+i3BvC4+7tZjZi1G1DlGAXumBXowDQ4X9HzJl+NrKW5rQ=
=xUu9
-----END PGP SIGNATURE-----

**Comment by scmschmidt** [comment]
Fixed with 1.0.3
